server/plugin/command.go

```diff
@@ -638,25 +652,20 @@ func getAutocompleteData(config *Configuration) *model.AutocompleteData {
 	subscriptionsAdd := model.NewAutocompleteData("add", "[owner/repo] [features] [flags]", "Subscribe the current channel to receive notifications about opened pull requests and issues for an organization or repository. [features] and [flags] are optional arguments")
 	subscriptionsAdd.AddTextArgument("Owner/repo to subscribe to", "[owner/repo]", "")
 	subscriptionsAdd.AddTextArgument("Comma-delimited list of one or more of: issues, pulls, pulls_merged, pushes, creates, deletes, issue_creations, issue_comments, pull_reviews, label:\"<labelname>\". Defaults to pulls,issues,creates,deletes", "[features] (optional)", `/[^,-\s]+(,[^,-\s]+)*/`)
-	if config.GitHubOrg != "" {
-		exclude := []model.AutocompleteListItem{
-			{
-				HelpText: "notifications for these repos will be turned off",
-				Hint:     "(optional)",
-				Item:     "--exclude",
-			},
-		}
-		subscriptionsAdd.AddStaticListArgument("Currently supports --exclude", true, exclude)
-		subscriptionsAdd.AddTextArgument("Owner/repo to subscribe to", "[owner/repo]", "")
-		flags := []model.AutocompleteListItem{
-			{
-				HelpText: "Events triggered by organization members will not be delivered (the organization config should be set, otherwise this flag has no effect)",
-				Hint:     "(optional)",
-				Item:     "--exclude-org-member",
-			},
-		}
-		subscriptionsAdd.AddStaticListArgument("Currently supports --exclude-org-member ", false, flags)
-	}
+	// if config.GitHubOrg != "" {
+	subscriptionsAdd.AddNamedStaticListArgument("", "Currently supports --exclude and --exclude-org-member", true, []model.AutocompleteListItem{
+		{
+			HelpText: "notifications for these repos will be turned off",
+			Hint:     "(optional)",
+			Item:     "--exclude",
+		},
+		{
+			HelpText: "Events triggered by organization members will not be delivered (the organization config should be set, otherwise this flag has no effect)",
+			Hint:     "(optional)",
+			Item:     "--exclude-org-member",
+		},
+	})
+	// }
```

Comment:
This won't produce the right autocomplete because we are supposed to have two different field types here. One is text-based, and one is boolean-based.

In order to support boolean-based arguments in our flag argument system, we will need to treat that field as a separate select element:

```go
const (
    // Defined at the top of the file
    commandArgumentExcludeHint = "Notifications for these repos will be turned off"
    commandArgumentExcludeOrgMemberHint = "Events triggered by organization members will not be delivered (default false)"
)

cmd.AddNamedTextArgument("exclude", commandArgumentExcludeHint, "", false)

if config.GitHubOrg != "" {
    cmd.AddNamedStaticListArgument("exclude-org-member", commandArgumentExcludeOrgMemberHint, false, []model.AutocompleteListItem{
        {
            Item:     "true",
            HelpText: "Exclude posts from members of the configured organization",
        }, {
            Item:     "false",
            HelpText: "Include posts from members of the configured organization",
        },
    })
}
```

Reply:
feedback point 1, remove GithubOrg, https://github.com/mattermost/mattermost-plugin-github/pull/505#issuecomment-1017705504
point 3 exclude and exclude-org-member is positional arguments. fixed it.

Reply:
i will check and update as feedback